### PR TITLE
Pinned `grcov` to v0.8.0

### LINKED ISSUE #369: GitHub actions implementation improvements
The current implementation of our buildscripts, do cache builds across commits in the same PR but the deployment step does not re-use the image generated in previous steps (verify that this is generated), making deployments very slow.

Figure out how to reduce the times we build the project.

- [x] improve safety of how coverge report is uploaded
- [x] improve the deployment time by reusing builds from previous steps

### PR DESCRIPTION
**Pull request by jpalvarezl**
Closes #369 

Changes:
- Pinned version of grcov to `v0.8.0`
- Removed any `spectrum` associated branch in the deployment script

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -116,7 +116,3 @@ jobs:
       - name: Deploy develop # unused
         if: github.ref == 'refs/heads/develop'
         run: bash scripts/deploy_docker.sh develop
```
Comment (fmrsabino) [inline .github/workflows/rust.yml:118]: This can also be removed right?

**Review by fmrsabino — APPROVED** [review]

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -68,11 +68,11 @@ jobs:
         uses: actions/cache@v2
```
Comment (fmrsabino) [inline .github/workflows/rust.yml:68]: Not sure if the scope of this PR is just for pinning `grcov` but we can also update the cache action (can be done in a different PR)

```suggestion
        uses: actions/cache@v2.1.5
```

Comment (jpalvarezl) [reply .github/workflows/rust.yml:68]: Thanks for that! Yeah, I don't see why we couldn't include those changes :)

**Review by fmrsabino — APPROVED** [review]

File: .github/workflows/rust.yml (hunk 2 above)
Comment (jpalvarezl) [reply .github/workflows/rust.yml]: just for reference: https://github.com/actions/cache/releases if we want to check release notes.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (jpalvarezl) [reply .github/workflows/rust.yml:118]: I think I would keep it, as it is not entirely unlikely we introduce a dev branch at some point

**Pull request merged**